### ISSUE TITLE
Multiple entries: Disallowing setting entries as "Finished" if they got validation warnings

### ISSUE THREAD

**Issue by andreoda** [VocaDB/vocadb#1833]
Users can set entries as "Finished" even if they have entry warnings.

For example:
https://vocadb.net/S/677428
Warning: This entry is missing the following information: Derived songs need to have an original version specified, if it's in the database, or the original needs to be mentioned in notes.

To reduce problems like these, the entries should only be closing as "Draft" status automatically (appear as the only option).

This feature/rule should only set in if the user is a "Regular". "Trusted" users and above should still be able to set entries as "Finished" due to exceptions, for example cover unifier entries.

**Comment by Susko3** [comment]
What should happen if a regular user tries to edit and save a "Finished" song that has entry warnings?

**Comment by andreoda** [comment]
@Susko3
As of now, nothing happens and everyone can save entries as "Finished" so if this feature gets added, the only options for regular users will be to save as "Draft", the other options will not be available for them as long as entry warnings are still active/appearing.

**Comment by Susko3** [comment]
So if a regular user tries to edit a forced "Finished" song (set as such by a trusted user), let's say they are fixing a typo in a cover unifier. The entry will "automatically" get downgraded to "Draft". That doesn't seem right to me.

If a trusted user has force-marked a song as finished, regular user edits shouldn't automatically downgrade the entry to draft status.

**Comment by andreoda** [comment]
@Susko3
There is also the option to introduce a rule that all entries with the cover unifier tag do not get the specific entry/validation warnings.

**Comment by CatgirlFrostmoon** [comment]
#1325 is the relevant "allow cover unifiers to bypass validation warnings" issue, and is pretty much the only current case where ignoring validation warnings is a common, acceptable, and even encouraged practice. 1325 would definitely have to be added to the database before this issue.

I will note that we have multiple open issues that talk about adding new validation warnings themselves, and all of them could make this issue less friendly to editors if implemented poorly: 
- #1368 wants this to appear if a person is set as both character designer and illustrator for a bank (may have problems if there are multiple illustrators / CDs, as is the case with [Kanade Shia](https://vocadb.net/Ar/87386))
- #1362 wants this to appear if a derived voicebank duplicates the same voice provider / illustrator / etc. as the main bank (which just generally runs into the issues inherent in our inheritance system, as I talk about in #2098)
- #1481 is one of the amalgamate issues, in this case acting as a unifier for all validation warning issues - but it also includes some new ones itself, a few of which could run into serious problems (i.e. the Chinese venue one, which might just lead to a whole slew of unfinishable entries...?)

Basically: this definitely isn't a bad idea, but anything that forcibly makes an entry unfinishable needs to handled with delicacy and thorough care.
